src/common/commonutils/CommandUtils.c

```diff
@@ -29,270 +31,200 @@ static int NormalizeStatus(int status)
+                OsConfigLogError(log, "Timeout reading from pipe for command '%s'", command);
                 status = ETIME;
-                if (NULL == callback)
-                {
-                    sleep(timeout);
-                }
-                else
-                {
-                    while (totalWaitSeconds < timeout)
-                    {
-                        // If the callback returns non zero, cancel the command
-                        if (0 != callback(context))
-                        {
-                            status = ECANCELED;
-                            break;
-                        }
-                        sleep(callbackIntervalSeconds);
-                        totalWaitSeconds += callbackIntervalSeconds;
-                    }
-                }
-                _exit(status);
+                break;
             }
-            else if (timerProcess < 0)
+            if ((NULL != callback) && (currentTime - lastCallbackTime >= callbackIntervalSeconds))
             {
-                // Intermediate process
-                OsConfigLogError(log, "Failed forking timer process (errno: %d, '%s')", errno, strerror(errno));
-                status = -1;
-
-                // Kill the worker process if created and wait on it before exiting otherwise worker becomes a zombie process
-                if (workerProcess > 0)
+                if (0 != callback(context))
                 {
-                    KillProcess(workerProcess);
-                    waitpid(workerProcess, &intermediateStatus, 0);
+                    OsConfigLogError(log, "Canceled reading from pipe for command '%s'", command);
+                    status = ECANCELED;
+                    break;
                 }
-                _exit(status);
+                lastCallbackTime = currentTime;
             }
 
-            // Wait on the child process (worker or timer) that finishes first
-            childProcess = waitpid(0, &status, 0);
-            status = NormalizeStatus(status);
-            if (childProcess == workerProcess)
+            if (!FD_ISSET(pipefd[0], &fdset))
             {
-                OsConfigLogDebug(log, "Command execution returning %d", status);
-                KillProcess(timerProcess);
+                // It was a timeout, nothing to read, loop.
+                continue;
             }
-            else
+
+            bytesRead = read(pipefd[0], buffer, BUFFER_SIZE);
+            if (bytesRead == 0)
             {
-                // Timer process is done, kill the timed out worker process
-                OsConfigLogDebug(log, "Command timed out or it was canceled, command process killed (%d)", status);
-                KillProcess(workerProcess);
+                // Child closed the pipe, we are done.
+                status = 0;
+                break;
+            }
+            if (bytesRead < 0)
+            {
+                if (EINTR == errno) {
+                    continue;
+                }
```

Comment:
Here and elsewhere, please use curly braces {} consistent with existing code. 

Also, here local var status is used. Can we please use this everywhere in the whole ExecuteCommand function?